client/rest/src/main/java/org/opensearch/client/RequestOptions.java

```diff
@@ -142,6 +155,12 @@ public String toString() {
                 b.append(headers.get(h).toString());
             }
         }
+        if (queryParams.size() > 0) {
```

Comment:
Please remove this code, this is not the place the parameters should be set (in my opinion). The `org.opensearch.client.Request` has the reference to `RequestOptions` associated with it and could provide the combined parameters set (own + options) when `getParameters` are called.

Reply:
@reta I'm a little confused by your comment here.  Did you realise that this code is part of the `toString()` method?  I have tried to reflect a similar form of `toString()` implementation as is used for outputting the headers.

Reply:
@oliverlockwood I think I didn't realize that (and haven't looked into context around this method), it should be fine, thank you for pointing it out.